src/rprblender/engine/viewport_engine.py

```diff
@@ -346,7 +352,10 @@ class FinishRender(Exception):
 
                     self.is_rendered = True
                     self.is_denoised = False
-                    iteration += 1
+                    iteration += num_samples
+                    if iteration > 1 and num_samples < 32:
```

Comment:
32 is just too much for viewport if scene is at least somewhat complex or has refractive objects, even 8 is not that comfortable.